tools/analysis_tools/confusion_matrix.py

```diff
@@ -75,27 +77,24 @@ def calculate_confusion_matrix(dataset,
         tp_iou_thr (float|optional): IoU threshold to be considered as matched.
             Default: 0.5.
     """
-    num_classes = len(dataset.CLASSES)
+    num_classes = len(dataset.metainfo['CLASSES'])
     confusion_matrix = np.zeros(shape=[num_classes + 1, num_classes + 1])
     assert len(dataset) == len(results)
     prog_bar = mmcv.ProgressBar(len(results))
     for idx, per_img_res in enumerate(results):
         if isinstance(per_img_res, tuple):
```

Comment:
there will be no such a case when the prediction results is a tuple, we should clean the logic.